catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisDatabaseOperations.java

```diff
@@ -4,28 +4,121 @@
+import com.datastrato.gravitino.meta.AuditInfo;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collections;
 import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
 
 /** Database operations for Doris. */
 public class DorisDatabaseOperations extends JdbcDatabaseOperations {
-  // TODO: add implementation for doris catalog
 
   @Override
-  public JdbcSchema load(String databaseName) throws NoSuchSchemaException {
-    throw new UnsupportedOperationException();
+  public String generateCreateDatabaseSql(
+      String databaseName, String comment, Map<String, String> properties) {
+    if (StringUtils.isNotEmpty(comment)) {
+      throw new UnsupportedOperationException(
+          "Doris doesn't support set database comment: " + comment);
+    }
+    StringBuilder sqlBuilder = new StringBuilder();
+
+    // Append database name
+    sqlBuilder.append(String.format("CREATE DATABASE `%s`", databaseName));
+
+    // Append properties
+    sqlBuilder.append(DorisUtils.generatePropertiesSql(properties));
+
+    String result = sqlBuilder.toString();
+    LOG.info("Generated create database:{} sql: {}", databaseName, result);
+    return result;
   }
 
   @Override
-  protected String generateCreateDatabaseSql(
-      String databaseName, String comment, Map<String, String> properties) {
-    throw new UnsupportedOperationException();
+  public String generateDropDatabaseSql(String databaseName, boolean cascade) {
+    StringBuilder sqlBuilder = new StringBuilder();
+    sqlBuilder.append(String.format("DROP DATABASE `%s`", databaseName));
+    if (cascade) {
+      sqlBuilder.append(" FORCE");
+      return sqlBuilder.toString();
+    }
+
+    try (final Connection connection = this.dataSource.getConnection()) {
+      String query = String.format("SHOW TABLES IN `%s`", databaseName);
+      try (Statement statement = connection.createStatement()) {
+        // Execute the query and check if there exists any tables in the database
+        try (ResultSet resultSet = statement.executeQuery(query)) {
+          if (resultSet.next()) {
+            throw new IllegalStateException(
+                String.format(
+                    "Database %s is not empty, the value of cascade should be true.",
+                    databaseName));
+          }
+        }
+      }
+    } catch (SQLException sqlException) {
+      throw this.exceptionMapper.toGravitinoException(sqlException);
+    }
+
+    return sqlBuilder.toString();
   }
 
   @Override
-  protected String generateDropDatabaseSql(String databaseName, boolean cascade) {
-    throw new UnsupportedOperationException();
+  public JdbcSchema load(String databaseName) throws NoSuchSchemaException {
+    try (final Connection connection = this.dataSource.getConnection()) {
+      String query = "SELECT * FROM information_schema.SCHEMATA WHERE SCHEMA_NAME = ?";
```

Comment:
We do not suggest using system table `information_schema.SCHEMATA`, please refer to method `JdbcTableOperations#load`

Reply:
Maybe we can only use `information_schema.SCHEMATA` for database information.

https://github.com/datastrato/gravitino/blob/db3633244bb6fc7dc7de12b59e0d0607318fedcd/catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java#L96

https://github.com/datastrato/gravitino/blob/db3633244bb6fc7dc7de12b59e0d0607318fedcd/catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlSchemaOperations.java#L54

Reply:
@zhoukangcn 
We are going to remove all places that depend on system table like https://github.com/datastrato/gravitino/pull/2616

Reply:
I see.  I will update my code when 2616 is merged

Reply:
done